### PR TITLE
feat: update weaviate server to v1.16.0 and client to 3.9.0

### PR DESCRIPTION
**Pull request by dirkkul**
Based on https://github.com/docarray/docarray/pull/734

Goals:
- [x] update weaviate server and weaviate client
- [x] update some python client calls to include class names
- [x] update some of the tests

### CODE REVIEW

File: tests/unit/array/mixins/test_find.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def inv_cosine(*args):
     'storage, config',
     [
         ('memory', None),
-        ('weaviate', {'n_dim': 32}),
+        ('weaviate', {'n_dim': 32, "distance": "cosine"}),
```
Comment (alaeddine-13) [inline tests/unit/array/mixins/test_find.py:32]:
```suggestion
        ('weaviate', {'n_dim': 32, 'distance': 'cosine'}),
```

File: tests/unit/array/mixins/test_find.py [hunk 2]
```diff
@@ -51,7 +51,10 @@ def test_find(storage, config, limit, query, start_storage):
 
     da.extend([Document(embedding=v) for v in embeddings])
 
-    result = da.find(query, limit=limit)
+    if storage == "weaviate":
```
Comment (alaeddine-13) [inline tests/unit/array/mixins/test_find.py:54]:
```suggestion
    if storage == 'weaviate':
```

File: tests/unit/array/mixins/test_find.py [hunk 3]
```diff
@@ -51,7 +51,10 @@ def test_find(storage, config, limit, query, start_storage):
...
+        result = da.find(query, limit=limit, additional=["certainty"])
```
Comment (alaeddine-13) [inline tests/unit/array/mixins/test_find.py:55]:
```suggestion
        result = da.find(query, limit=limit, additional=['certainty'])
```

File: tests/unit/array/storage/weaviate/test_query_params.py [hunk 4]
```diff
@@ -6,13 +6,14 @@ def find_random(da, target_certainty):
     return da.find(
         DocumentArray([Document(embedding=np.random.randint(10, size=10))]),
         query_params={"certainty": target_certainty},
+        additional=["certainty"],
```
Comment (alaeddine-13) [inline tests/unit/array/storage/weaviate/test_query_params.py:9]:
```suggestion
        additional=['certainty'],
```

File: tests/unit/array/storage/weaviate/test_query_params.py [hunk 5]
```diff
@@ -6,13 +6,14 @@ def find_random(da, target_certainty):
...
     )[0]
 
 
 def test_certainty_filter(start_storage):
     nrof_docs = 100
     target_certainty = 0.98
-    da = DocumentArray(storage="weaviate", config={"n_dim": 10})
+    da = DocumentArray(storage="weaviate", config={"n_dim": 10, "distance": "cosine"})
```
Comment (alaeddine-13) [inline tests/unit/array/storage/weaviate/test_query_params.py:16]:
```suggestion
    da = DocumentArray(storage='weaviate', config={'n_dim': 10, 'distance': 'cosine'})
```

**Review by alaeddine-13 — CHANGES_REQUESTED** [review]

File: docarray/array/storage/weaviate/getsetdel.py [hunk 6]
```diff
@@ -61,8 +63,8 @@ def _del_doc_by_id(self, _id: str):
         :param _id: the id of the document to delete
         """
         wid = self._map_id(_id)
-        if self._client.data_object.exists(wid):
-            self._client.data_object.delete(wid)
+        if self._client.data_object.exists(wid, class_name=self.name):
+            self._client.data_object.delete(wid, class_name=self.name)
```
[the remark this answers is not in the document]
Comment (dirkkul) [reply docarray/array/storage/weaviate/getsetdel.py]: please have a look if it's ok!

[the remark this answers is not in the document]
Comment (alaeddine-13) [reply docarray/array/storage/weaviate/getsetdel.py]: looks good to me, let's try to pass CI

**Review by alaeddine-13 — APPROVED** [review]

**Pull request merged**